src/main/java/uk/gov/hmcts/cmc/claimstore/events/DocumentUploadHandler.java

```diff
@@ -1,43 +1,163 @@
+        this.countyCourtJudgmentPdfService = countyCourtJudgmentPdfService;
+        this.settlementAgreementCopyService = settlementAgreementCopyService;
+        this.sealedClaimPdfService = sealedClaimPdfService;
+        this.claimIssueReceiptService = claimIssueReceiptService;
+        this.defendantPinLetterPdfService = defendantPinLetterPdfService;
+        this.documentService = documentService;
+    }
+
+    @EventListener
+    public void uploadDocument(CitizenClaimIssuedEvent event) {
+        Claim claim = event.getClaim();
+        requireNonNull(claim, CLAIM_MUST_NOT_BE_NULL);
+        PDF sealedClaim = new PDF(buildSealedClaimFileBaseName(event.getClaim().getReferenceNumber()),
+            sealedClaimPdfService.createPdf(event.getClaim()),
+            SEALED_CLAIM);
+        PDF defendantLetter = new PDF(buildDefendantLetterFileBaseName(event.getClaim().getReferenceNumber()),
+            defendantPinLetterPdfService.createPdf(event.getClaim(), event.getPin()),
+            DEFENDANT_PIN_LETTER);
+        PDF claimIssueReceipt = new PDF(buildClaimIssueReceiptFileBaseName(event.getClaim().getReferenceNumber()),
+            claimIssueReceiptService.createPdf(event.getClaim()),
+            CLAIM_ISSUE_RECEIPT);
+        uploadToDocumentManagement(event.getClaim(),
+            event.getAuthorisation(),
+            sealedClaim,
+            defendantLetter,
+            claimIssueReceipt);
+    }
+
+    @EventListener
+    public void uploadDocument(RepresentedClaimIssuedEvent event) {
+        Claim claim = event.getClaim();
+        requireNonNull(claim, CLAIM_MUST_NOT_BE_NULL);
+        PDF sealedClaim = new PDF(buildSealedClaimFileBaseName(event.getClaim().getReferenceNumber()),
+            sealedClaimPdfService.createPdf(event.getClaim()),
+            SEALED_CLAIM);
+        uploadToDocumentManagement(event.getClaim(),
+            event.getAuthorisation(),
+            sealedClaim);
     }
 
     @EventListener
     public void uploadDocument(DefendantResponseEvent event) {
         Claim claim = event.getClaim();
-        requireNonNull(claim, "Claim must be present");
+        requireNonNull(claim, CLAIM_MUST_NOT_BE_NULL);
         if (!claim.getResponse().isPresent()) {
             throw new IllegalArgumentException("Response must be present");
         }
         PDF defendantResponseDocument = new PDF(buildResponseFileBaseName(claim.getReferenceNumber()),
             defendantResponseReceiptService.createPdf(claim),
             DEFENDANT_RESPONSE_RECEIPT);
-        publisher.publishEvent(new DocumentUploadEvent(claim,
-            event.getAuthorization(),
-            defendantResponseDocument));
+        uploadToDocumentManagement(claim, event.getAuthorization(), defendantResponseDocument);
+    }
+
+    @EventListener
+    public void uploadDocument(CountyCourtJudgmentEvent event) {
+        Claim claim = event.getClaim();
+        requireNonNull(claim, CLAIM_MUST_NOT_BE_NULL);
```

Comment:
Suggestion : Do we need to do additional check if the CCJ related details are existing before proceeding further? Same, check other methods too to verify if the claim is in appropriate state.

Reply:
the controller already checks this condition.